src/openfermion/transforms/_code_transform_functions.py

```diff
@@ -0,0 +1,238 @@
+
+    Args:
+        modes (int):  number of modes
+        odd (int or bool): 1 (True) or 0 (False), if odd,
+            we encode all states with odd Hamming weight
+
+    Returns (list): list of SymbolicBinary
+    """
+    if odd:
+        all_in = SymbolicBinary('1')
+    else:
+        all_in = SymbolicBinary()
+
+    for mode in range(modes - 1):
+        all_in += SymbolicBinary('w' + str(mode))
+
+    djw = linearize_decoder(numpy.identity(modes - 1, dtype=int))
+    djw.append(all_in)
+    return djw
+
+
+def checksum_code(modes, odd):
+    """ Checksum code for either even or odd Hamming-weight
+        
+    Args:
+        modes (int): number of modes
+        odd (int or bool): 1 (True) or 0 (False), if odd,
+            we encode all states with odd Hamming weight
+
+    Returns (BinaryCode): The checksum BinaryCode
+    """
+    return BinaryCode(_encoder_checksum(modes), _decoder_checksum(modes, odd))
+
+
+def jordan_wigner_code(modes):
+    """ The Jordan-Wigner transform as binary code.
+        
+    Args:
+        modes (int): number of modes
+
+    Returns (BinaryCode): The Jordan-Wigner BinaryCode
+    """
+    return BinaryCode(numpy.identity(modes, dtype=int), linearize_decoder(
+        numpy.identity(modes, dtype=int)))
+
+
+def bravyi_kitaev_code(modes):
+    """ The Bravyi-Kitaev transform as binary code.
+        
+    Args:
+        modes (int): number of modes
+
+    Returns (BinaryCode): The Bravyi-Kitaev BinaryCode
+    """
+    return BinaryCode(_encoder_bk(modes),
+                      linearize_decoder(_decoder_bk(modes)))
+
+
+def parity_code(modes):
+    """ The parity transform as binary code.
+        
+    Args:
+        modes (int): number of modes
+
+    Returns (BinaryCode): The parity transform BinaryCode
+    """
+    dec_mtx = numpy.reshape(([1] + [0] * (modes - 1)) +
+                            ([1, 1] + (modes - 1) * [0]) * (modes - 2) +
+                            [1, 1], (modes, modes))
+    enc_mtx = numpy.tril(numpy.ones((modes, modes), dtype=int))
+
+    return BinaryCode(enc_mtx, linearize_decoder(dec_mtx))
+
+
+def _binary_address(digits, address):
+    """ Helper function to fill in an encoder column/decoder component of a
+    certain number.
+
+    Args:
+        digits (int): number of digits, which is the qubit number
+        address (int): column index, decoder component
+
+    Returns (tuple): encoder column, decoder component
+    """
+    binary_expression = SymbolicBinary('1')
+
+    # isolate the binary number and fill up the mismatching digits
+    address = bin(address)[2:]
+    address = ('0' * (digits - len(address))) + address
+    for index in numpy.arange(digits):
+        binary_expression *= SymbolicBinary(
+            'w' + str(index) + ' + 1 + ' + address[index])
+
+    return list(map(int, list(address))), binary_expression
+
+
+def weight_one_binary_addressing_code(exponent):
+    """ Weight-1 binary addressing code.
+
```

Comment:
Again, it would be good to have some more explanation of what these codes are and why people might be interested in them.